engine/runtime-integration-tests/src/test/java/org/enso/compiler/ImportsAndFQNConsistencyTest.java

```diff
@@ -0,0 +1,230 @@
+package org.enso.compiler;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.enso.compiler.core.ir.module.scope.Definition;
+import org.enso.compiler.core.ir.module.scope.Export;
+import org.enso.interpreter.runtime.EnsoContext;
+import org.enso.interpreter.test.TestBase;
+import org.enso.polyglot.LanguageInfo;
+import org.enso.polyglot.RuntimeOptions;
+import org.graalvm.polyglot.Context;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+import org.junit.runners.model.Statement;
+
+/**
+ * If a symbol is importable by an import statement, then it should also be accessible via FQN
+ * without any import (technically, with just an import of the library). This rule is enforced by
+ * this test. It uses symbols from standard libraries. It scans their Main modules for exported
+ * symbols, and then tests whether this invariant holds for all the symbols. Note that not all the
+ * symbols are tested. For example the builtin types are not tested.
+ */
+@RunWith(Parameterized.class)
+public class ImportsAndFQNConsistencyTest extends TestBase {
+  private static Context ctx;
+
+  /** Used for description in {@link PrintCodeRule} test rule. */
+  private static String code;
+
+  @Rule public final TestRule printCodeRule = new PrintCodeRule();
+
+  /**
+   * Gather all the exported symbols from Standard.Base library. Exclude builtin symbols, and
+   * modules with extension methods.
```

Comment:
Why are modules with extension methods excluded?

Or does that just mean that we exclude modules that define _only_ ext methods and thus export no types on their own?

Reply:
> Why are modules with extension methods excluded?

This is just a limitation of my original testing code. I was not sure how to call extension methods properly directly, for example, from `Standard.Base.Data.Test.Extensions`. Any suggestions?

Note that this testing class is still a work-in-progress. Despite Jaroslav marking this PR as ready-for-review.

Reply:
The goal of this PR isn't to write a full featured set of tests, but to make sure _imports and FQNs_ behave the same in the runtime. I am using relevant @Akirathan tests from his #9367 to verify the _imports and FQNs consistency_ improves.

Given we now pass 180 new tests, the **consistency has clearly improved**. If there is a know case where FQNs and imports aren't consistent, let's write a test for it, but _increasing coverage just to increase coverage_ isn't goal of this PR.